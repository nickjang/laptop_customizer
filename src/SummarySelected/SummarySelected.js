import React, { Component } from 'react';
import USCurrencyFormat from '../USCurrencyFormat'

import './SummarySelected.css'

class SummarySelected extends Component {
  static defaultProps = {
    feature: [],
    selectedOption: {}
  };

  render() {
    const selectedOption = this.props.selectedOption;
    return (
      <div className="summary__option" >
        <div className="summary__option__label">{this.props.feature} </div>
        <div className="summary__option__value">{selectedOption.name}</div>
        <div className="summary__option__cost">
          {USCurrencyFormat.format(selectedOption.cost)}
        </div>
      </div>
    );
  }
}

export default SummarySelected;