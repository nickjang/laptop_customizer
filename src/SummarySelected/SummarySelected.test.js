import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SummarySelected from './SummarySelected';

describe('<SummarySelected />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SummarySelected />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<SummarySelected />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders selected option when props are supplied', () => {
    const wrapper = shallow(
      <SummarySelected feature="Display" idx={1} 
        selectedOption={{
          name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
          cost: 1500
        }} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});