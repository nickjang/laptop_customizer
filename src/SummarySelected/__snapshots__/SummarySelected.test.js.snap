// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummarySelected /> renders empty when no props supplied 1`] = `
<div
  className="summary__option"
>
  <div
    className="summary__option__label"
  >
     
  </div>
  <div
    className="summary__option__value"
  />
  <div
    className="summary__option__cost"
  >
    $NaN
  </div>
</div>
`;

exports[`<SummarySelected /> renders selected option when props are supplied 1`] = `
<div
  className="summary__option"
>
  <div
    className="summary__option__label"
  >
    Display
     
  </div>
  <div
    className="summary__option__value"
  >
    15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs
  </div>
  <div
    className="summary__option__cost"
  >
    $1,500.00
  </div>
</div>
`;
